src/dialects/mssql/query-interface.js

```diff
@@ -50,6 +50,19 @@ export class MsSqlQueryInterface extends AbstractQueryInterface {
     return this.sequelize.queryRaw(removeSql, options);
   }
 
+  /**
+    * @override
+    */
+  async bulkInsert(tableName, records, options, attributes) {
+    // If more than 1,000 rows are inserted outside of a transaction, we can't guarantee safe rollbacks.
+    // See https://github.com/sequelize/sequelize/issues/15426
+    if (records.length > 1000 && !options.transaction) {
+      throw new Error(`MSSQL doesn't allow for inserting more than 1,000 rows at a time, so Sequelize executes the insert as multiple queries. Please run this in a transaction to ensure safe rollbacks`);
+    }
+
+    super.bulkInsert(tableName, records, options, attributes);
```

Comment:
```suggestion
    await super.bulkInsert(tableName, records, options, attributes);
```
This could fix it, right?